File: src/test/java/it/marketplace/microservices/rabbitmq/MyRabbitListenerTest.java
package it.marketplace.microservices.rabbitmq;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import java.util.HashMap;
import java.util.Map;

import static org.mockito.Mockito.verify;

class MyRabbitListenerTest {

    @Mock
    private TransactionService service;

    @InjectMocks
    private MyRabbitListener listener;

    @BeforeEach
    void setUp() {
        MockitoAnnotations.openMocks(this);
    }

    @Test
    void shouldCallStartProcessing_WhenListenerNewOrder_ThenArrangeActAssert() throws Exception {
        // Arrange
        String orderCode = "ORD123";
        // Act
        listener.listenerNewOrder(orderCode);
        // Assert
        verify(service).startProcessing(orderCode);
    }

    @Test
    void shouldCallStartPendingPayment_WhenListenerPendingPayment_ThenArrangeActAssert() throws Exception {
        // Arrange
        Map<String, String> message = new HashMap<>();
        message.put("orderCode", "ORD123");
        // Act
        listener.listenerPendingPayment(message);
        // Assert
        verify(service).startPendingPayment(message);
    }
}

File: src/test/java/it/marketplace/microservices/rabbitmq/MyRabbitListenerTest.kt
package it.marketplace.microservices.rabbitmq

import it.marketplace.microservices.service.TransactionService
import it.marketplace.microservices.utils.BaseTest
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.mockito.InjectMocks
import org.mockito.Mock
import org.mockito.Mockito
import org.mockito.Mockito.mock
import org.mockito.MockitoAnnotations

class MyRabbitListenerTest : BaseTest() {

    @Mock
    private val service: TransactionService = mock()

    @InjectMocks
    private val listener: MyRabbitListener = mock()

    @BeforeEach
    fun setUp() {
        MockitoAnnotations.openMocks(this)
    }

    @Test
    @Throws(Exception::class)
    fun shouldCallStartProcessing_WhenListenerNewOrder_ThenArrangeActAssert() {
        // Arrange
        val orderCode = "ORD123"
        // Act
        listener.listenerNewOrder(orderCode)
        // Assert
        Mockito.verify(service).startProcessing(orderCode)
    }

    @Test
    @Throws(Exception::class)
    fun shouldCallStartPendingPayment_WhenListenerPendingPayment_ThenArrangeActAssert() {
        // Arrange
        val message = mapOf("orderCode" to "ORD123")
        // Act
        listener.listenerPendingPayment(message)
        // Assert
        Mockito.verify(service).startPendingPayment(message)
    }
}

